python/pyarrow/_compute.pyx

```diff
@@ -1331,6 +1331,32 @@ class Utf8NormalizeOptions(_Utf8NormalizeOptions):
         self._set_options(form)
 
 
+cdef class _RandomOptions(FunctionOptions):
+    def _set_options(self, length, initializer):
+        if initializer == 'system':
+            self.wrapped.reset(new CRandomOptions(
+                CRandomOptions.FromSystemRandom(length)))
+            return
+
+        if not isinstance(initializer, int):
+            try:
+                initializer = hash(initializer)
+            except TypeError:
+                raise TypeError(
+                    f"initializer should be 'system', an integer, "
+                    f"or a hashable object; got {initializer!r}")
```

Comment:
Nit: Only the bottom string needs formatting. Similarly, in an error message below. Maybe you have it like this for symmetry or text alignment ;)

Reply:
Yes, and it's also less error-prone if we add some formatting to the top string.

Reply:
Is it useful to `raise from` here to preserve the original context?

Reply:
No, because Python automatically propagates the context. `raise from` adds a `__cause__`, which isn't very useful in practice.